src/ImageSharp/Formats/Webp/Lossy/YuvConversion.cs

```diff
@@ -312,6 +602,175 @@ public static void YuvToBgr(int y, int u, int v, Span<byte> bgr)
+                PlanarTo24Shuffle5,
+                out Vector128<byte> g0,
+                out Vector128<byte> g1,
+                out Vector128<byte> g2,
+                out Vector128<byte> g3,
+                out Vector128<byte> g4,
+                out Vector128<byte> g5);
+
+            // Process B.
+            ChannelMixing(
+                input4,
+                input5,
+                PlanarTo24Shuffle6,
+                PlanarTo24Shuffle7,
+                PlanarTo24Shuffle8,
+                out Vector128<byte> b0,
+                out Vector128<byte> b1,
+                out Vector128<byte> b2,
+                out Vector128<byte> b3,
+                out Vector128<byte> b4,
+                out Vector128<byte> b5);
+
+            // OR the different channels.
+            Vector128<byte> rg0 = Sse2.Or(r0, g0);
+            Vector128<byte> rg1 = Sse2.Or(r1, g1);
+            Vector128<byte> rg2 = Sse2.Or(r2, g2);
+            Vector128<byte> rg3 = Sse2.Or(r3, g3);
+            Vector128<byte> rg4 = Sse2.Or(r4, g4);
+            Vector128<byte> rg5 = Sse2.Or(r5, g5);
+
+            ref byte outputRef = ref MemoryMarshal.GetReference(rgb);
+            Unsafe.As<byte, Vector128<byte>>(ref outputRef) = Sse2.Or(rg0, b0);
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, 16)) = Sse2.Or(rg1, b1);
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, 32)) = Sse2.Or(rg2, b2);
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, 48)) = Sse2.Or(rg3, b3);
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, 64)) = Sse2.Or(rg4, b4);
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, 80)) = Sse2.Or(rg5, b5);
+        }
+
+        // Shuffles the input buffer as A0 0 0 A1 0 0 A2
+        private static void ChannelMixing(
+            Vector128<byte> input0,
+            Vector128<byte> input1,
+            Vector128<byte> shuffle0,
+            Vector128<byte> shuffle1,
+            Vector128<byte> shuffle2,
+            out Vector128<byte> output0,
+            out Vector128<byte> output1,
+            out Vector128<byte> output2,
+            out Vector128<byte> output3,
+            out Vector128<byte> output4,
+            out Vector128<byte> output5)
+        {
+            output0 = Ssse3.Shuffle(input0, shuffle0);
+            output1 = Ssse3.Shuffle(input0, shuffle1);
+            output2 = Ssse3.Shuffle(input0, shuffle2);
+            output3 = Ssse3.Shuffle(input1, shuffle0);
+            output4 = Ssse3.Shuffle(input1, shuffle1);
+            output5 = Ssse3.Shuffle(input1, shuffle2);
+        }
+
+        // Convert 32 samples of YUV444 to R/G/B
+        private static void ConvertYuv444ToRgbSse41(Span<byte> y, Span<byte> u, Span<byte> v, out Vector128<short> r, out Vector128<short> g, out Vector128<short> b)
+        {
+            Vector128<byte> y0 = LoadHigh(ref MemoryMarshal.GetReference(y));
```

Comment:
Ignore me if you already double checked this, but what you really need to make sure now is that if `y`,`u`,`v` have 8 bytes even at the very end, otherwise it's a memory corruption.

The code that allocates these buffers seems to add padding at first glance, but I don't really follow it:

https://github.com/SixLabors/ImageSharp/blob/099a67605e36262601101c5ef51f7fcea3ba9962/src/ImageSharp/Formats/Webp/Lossy/Vp8Decoder.cs#L66-L73